x-pack/plugins/reporting/server/browsers/chromium/driver/screenshot_stitcher/index.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { map, mergeMap, switchMap, toArray } from 'rxjs/operators';
+import { $combine } from './combine';
+import { $getClips } from './get_clips';
+import { Dimension, Logger, Screenshot } from './types';
+
+const scaleRect = (rect: Dimension, scale: number): Dimension => {
+  return {
+    height: rect.height * scale,
+    width: rect.width * scale,
+    x: rect.x * scale,
+    y: rect.y * scale,
+  };
+};
+
+/**
+ * Returns a stream of data that should be of the size outputClip.width * zoom x outputClip.height * zoom
+ * @param outputClip - The dimensions the final image should take.
+ * @param zoom - Determines the resolution want the final screenshot to take.
+ * @param maxDimensionPerClip - The maximimum dimension, in any direction (width or height) that we should allow per
+ * screenshot clip. If zoom is 10 and maxDimensionPerClip is anything less than or
+ * equal to 10, each clip taken, before  being zoomed in, should be no bigger than 1 x 1.
+ * If zoom is 10 and maxDimensionPerClip is 20, then each clip taken before being zoomed in should be no bigger than 2 x 2.
+ * @param captureScreenshotFn - a function to take a screenshot from the page using the dimensions given. The data
+ * returned should have dimensions not of the clip passed in, but of the clip passed in multiplied by zoom.
```

Comment:
This is where the bug is I believe for #19563 - for some reason the captureScreenshotFn is returning data that is too small.  We may need to wrap it in a retry, since it seems flaky.  Ideally I can figure out why though...